src/main/flight/pid.c

```diff
@@ -631,6 +631,13 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
             detectAndSetCrashRecovery(pidProfile->crash_recovery, axis, currentTimeUs, delta, errorRate);
 
             pidData[axis].D = pidCoefficient[axis].Kd * delta * tpaFactor;
+            
+            if (gyroYawSpinDetected() )  {
+                // zero PIDs on pitch and roll leaving yaw P to correct spin 
+                pidData[axis].P = 0;
+                pidData[axis].I = 0;
+                pidData[axis].D = 0;
+            }
```

Comment:
Where did the code go that zeros the PIDs during gyro overflow detection?  In a true ICM overflow condition we can't leave yaw P active as the gyro may have inverted.  Also the logic to zero the PIDs when pid stabilization is not enabled is missing.

Reply:
It's right up at the start of the PID loop, around 540:

```
    if (!pidStabilisationEnabled || gyroOverflowDetected()) {
        for (int axis = FD_ROLL; axis <= FD_YAW; ++axis) {
            pidData[axis].P = 0;
            pidData[axis].I = 0;
            pidData[axis].D = 0;

            pidData[axis].Sum = 0;
        }

```